Dockerfile

```diff
@@ -1,7 +1,7 @@
-FROM python:3.10-slim AS wheel-builder
```

Comment:
lets leave the default mlserver docker image that we build as part of the CI at python 3.10, if we want a docker image for different python version then the options are:
- parameterise this `Dockerfile` to take the python version as argument.
- provide instructions on how to build a docker image with a different python version

In any case I think this should be done as a follow up PR perhaps.